umm-spec-lib/src/cmr/umm_spec/test/umm_generators.clj

```diff
@@ -34,12 +34,24 @@
       (throw (Exception. (format "The fields [%s] are not supported by generators with schema type [%s]"
                                  (pr-str unexpected-fields) (pr-str schema-type)))))))
 
+(defn- add-dependencies
+  "Checks if property has a dependency, if the dependency is not present it will add it
+   to the selected-properties collection"
+  [dependencies selected-properties]
+  (reduce
+   (fn [properties property]
+     (vec (set/union (set properties)
+                     (set (map keyword (property dependencies)))
```

Comment:
I think we're expecting `property` to be a keyword? If that's true I'd change this to `(get dependencies property)` so that it's clear property is not a function.